roles/openshift_adm/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Ensure required variables are defined.
+  when: not lookup('ansible.builtin.vars', item)
+  ansible.builtin.fail:
+    msg: "Missing required parameter {{ item }}"
+  loop:
+    - cifmw_basedir
+    - cifmw_path
+    - cifmw_openshift_api
+    - cifmw_openshift_user
+    - cifmw_openshift_password
+    - cifmw_openshift_kubeconfig
+
+- name: Prepare and shutdown the OpenShift cluster.
+  when:
+    - cifmw_openshift_adm_op == 'shutdown'
+  ansible.builtin.include_tasks:
+    file: shutdown.yml
+
+- name: Wait for the OpenShift cluster to become stable.
+  when:
+    - cifmw_openshift_adm_op == 'stable'
```

Comment:
so you'd have to invoke the role 2x once for shutdown and then again for stable?
perhaps it would be useful for both to execute in one invocation or no?

Reply:
No exactly. However you assessment is correct. For the first run, yes, we would be waiting for the cluster twice. Once for generating the api certificate and then other post shutdown. 

The subsequent runs, we would only when the system is being powered ON.

`cifmw_openshift_adm_op` determines the call hence when the role is consumed. Only one op is executed.